src/Access/RoleCache.cpp

```diff
@@ -56,13 +56,29 @@ namespace
 }
 
 
-RoleCache::RoleCache(const AccessControl & access_control_)
-    : access_control(access_control_), cache(600000 /* 10 minutes */) {}
+RoleCache::RoleCache(const AccessControl & access_control_, int expiration_time)
+    : access_control(access_control_), cache(expiration_time * 1000 /* 10 minutes by default*/)
+{
+    LOG_TRACE(&Poco::Logger::get("RoleCache ()"), "ctor: expiration time is {}", expiration_time);
+}
 
 
 RoleCache::~RoleCache() = default;
 
 
+bool RoleCache::roleEnabled(const UUID & role_id)
+{
```

Comment:
Does this function assume that `mutex` is already locked?

Reply:
Good point. Simple answer - yes (and I'll add comment), 
but I have to ensure that callees are safe.